src/accel/TrackingManagerIntegration.cc

```diff
@@ -156,9 +158,16 @@ void TrackingManagerIntegration::BeginOfRunAction(G4Run const*)
         // Set tracking manager on workers when Celeritas is not fully disabled
         CELER_LOG(debug) << "Verifying tracking manager";
 
+        auto const& opts
+            = detail::IntegrationSingleton::instance().setup_options();
+        auto const offload_particles
+            = opts.offload_particles.empty()
+                  ? TrackingManagerConstructor::DefaultOffloadParticles()
+                  : TrackingManagerConstructor::UserOffloadParticles();
```

Comment:
This is pretty awkward since it reconstructs the offload particles again. How about having `TrackingManagerConstructor`'s `ConstructParticle` method setting some static data in `TrackingManagerIntegration` ? It might be better to have a conversation about this.